### ISSUE TITLE
GIFs

### ISSUE THREAD

**Issue by josecarlos2121** [HashLips/generative-art-node#66]
Hi there,

Really nice code!
I want to make some animated NFTs using GIFs, but unfortunately the output files always make only PNGs.
Is there a way to work with GIFs?

Thanks

**Comment by Admeister123** [comment]
Same question :)

**Comment by cubecreatures** [comment]
Me three, this would be an awesome feature.

**Comment by juschief** [comment]
Looking for this as well.  Basically more in depth layering options.  Something like how faticorns NFT project is in this article https://link.medium.com/Gf2mUpPmikb

**Comment by msqar** [comment]
Same question. Any ideas on how to support this?

**Comment by Viduum** [comment]
Yes, i want to make animated NFTs using GIFs and PNG,

How can i make it? 

The output is only PNG and there no animations,

can someone help me?

**Comment by DigitalAssetUniverse** [comment]
Damn why hasnt this been answered yet? Seems like it could be a quick fix.

**Comment by Viduum** [comment]
yes? How can you fix it?

I give PNG and GIFs inside and become GIFs that are animated?

**Comment by msqar** [comment]
It seems it won't be added anytime soon, there's a tool to do this (webapp) that supports gifs and even mp4 videos on layers. But it's paid, it could be done by the same owners that worked on this library. So if the gif support is released for free, they would be shooting to their own knees.

**Comment by LateButEarly** [comment]
@msqar which one?

**Comment by jamesryan-dev** [comment]
aren't .gifs now a part of the hashilips package?

**Comment by mograph** [comment]
No only sequencing layers not different layers by gifs to my knowledge

**Comment by jamesryan-dev** [comment]
yeah apologies I recently realised I had mis-read and presumed because you can export the collection as a single promo .gif that meant .gif's was a thing

**Comment by MeetanshiGraphics** [comment]
I think in near future they might introduce it. It's needed now.. and whatever they have created yet now.. is just fantastic.. so hoping for fantastic again..

**Comment by jalagar** [comment]
Hi all, currently https://github.com/MichaPipo/Generative_Gif_Engine works for generating gifs. It works by taking in spritesheets -> gifs. However, I have found issues with MichaPipo's javascript method of generating gifs by using the js libraries and navigating pixel by pixel where sometimes the colors are wrong. It is also very slow (4 minutes to generate 15 gifs at the highest quality) and take significant memory and CPU. I am working on expanding on MichaPipo's repo to work for separate png -> spritesheet -> rarity -> png -> gif. I switched to using [python PIL](https://github.com/python-pillow/Pillow) and I can generate 100 gifs in under 30s without losing quality.

I plan to write a medium post and open source the repo later this week and will share it here! Hopefully as a v2 I can incorporate nftchef's https://github.com/nftchef/art-engine logic to work for if-then statements and nested layer structure! Feel free to follow me on twitter https://twitter.com/jalagar_eth for updates as well.

**Comment by jalagar** [comment]
@josecarlos2121 @Admeister123 @cubecreatures @juschief @msqar @MeetanshiGraphics As promised I was able to build a tool to do exactly this! it takes png layers and turns them into generative gifs. Its forked from an earlier version of hashlips but I plan to update it to a more recent version soon.

Here's a medium article: https://jalagar-eth.medium.com/how-to-create-generative-animated-nft-art-in-under-an-hour-e7dab1785c56 on how it works.
And here's the repo: https://github.com/jalagar/Generative_Gif_Engine
Feel free to follow me on twitter: https://twitter.com/jalagar_eth for more updates. DM on Twitter or create an issue/discussion on the repo for a faster response.

You can see all 1k of them on https://opensea.io/collection/genesis-bouncing-ball. Let me know if you have any questions!

**Comment by whiskymiko** [comment]
hey @jalagar,
the Opensea collection looks pretty cool! Does your tool work with Solana, or only Eth?

**Comment by jalagar** [comment]
Hi @whiskymiko yes it will work on any chain, the tool just generates the gifs! You will need to upload the gifs to somewhere off chain and link it in your smart contract on Solana or any smart contract.

**Comment by whiskymiko** [comment]
Got it, thanks!

**Comment by jalagar** [comment]
Hi @whiskymiko I realized while looking deeper that Solana metadata is slightly configured differently. I will investigate on making the tool compatible with SOL, lmk if you have questions

**Comment by RageQuitnft** [comment]
Hi @jalagar ! Thanks so much for the code! I’m really sorry to ask this- but I’m a complete noob in coding and GitHub  :) - what file do I need to download? I’ll be using visual studio. I’ve used hash lips code many times, and I had to download a zip file and then unzip it. Again, super sorry for the noob question…

**Comment by jalagar** [comment]
Hi @RageQuitnft no need to apologize! We're all at different stages. You can go to https://github.com/jalagar/Generative_Gif_Engine and click on code and download the zip like so. You will need to download Python (which should already be downloaded on your computer) and node (which you should already have from using hashlips). Please add a question to https://github.com/jalagar/Generative_Gif_Engine/discussions if you have any issues I will reply there for visibility!
<img width="405" alt="Screen Shot 2022-04-08 at 10 04 44 AM" src="https://user-images.githubusercontent.com/98709501/162452065-48f48b67-2310-4d75-b220-8456fdafe22a.png">
